Source/WebCore/page/InteractionRegion.cpp

```diff
@@ -205,6 +235,35 @@ static bool cachedImageIsPhoto(const CachedImage& cachedImage)
     return true;
 }
 
+static RefPtr<Image> findIconImage(const RenderObject& renderer)
+{
+    if (const auto& renderImage = dynamicDowncast<RenderImage>(renderer)) {
+        if (!renderImage->cachedImage() || renderImage->cachedImage()->errorOccurred())
+            return nullptr;
+
+        auto* image = renderImage->cachedImage()->image();
+        if (!image)
+            return nullptr;
+
+        if (image->isSVGImage()
+            || (image->isBitmapImage() && image->nativeImage() && image->nativeImage()->hasAlpha()))
+            return image;
+    }
+
+    return nullptr;
+}
+
+static std::optional<std::pair<SVGSVGElement*, SVGGraphicsElement*>> findSVGClipElements(const RenderObject& renderer)
```

Comment:
This should likely not be returning a pair of raw pointers. It is risky security-wise. We likely want to have RefPtrs or WeakPtrs instead of raw pointers since the caller cannot easily assign the raw pointers to WeakPtr/RefPtr.

Reply:
Looking at the rest of the code, RefPtr seems appropriate.

Reply:
Thanks for the ... pointer :)